File: 0704-binary-search/0704-binary-search.py
from typing import List

class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums)  

        while l < r:
            m = l + (r - l) // 2  
            if nums[m] >= target:
                r = m  # 왼쪽으로 범위를 좁힘
            else:
                l = m + 1  # 오른쪽으로 범위를 좁힘

        if l < len(nums) and nums[l] == target:
            return l
        return -1
        

File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int start = 0;
        int end = nums.length - 1;

        while (start <= end) {
            int i = start + (end - start) / 2;

            if (nums[i] == target) {
                return i; 
            }

            if (nums[i] > target) {
                end = i - 1; 
            } else {
                start = i + 1; 
            }
        }

        return -1; 
    }
}